libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_IP.c

```diff
@@ -1894,6 +1907,129 @@ uint8_t ucProtocol;
 #endif /* ( ipconfigREPLY_TO_INCOMING_PINGS == 1 ) || ( ipconfigSUPPORT_OUTGOING_PINGS == 1 ) */
 /*-----------------------------------------------------------*/
 
+#if( ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM == 1 )
+	/* Although the driver will take care of checksum calculations,
+	the IP-task will still check if the length fields are OK. */
+	static BaseType_t xCheckSizeFields( const uint8_t * const pucEthernetBuffer, size_t uxBufferLength )
+	{
+	size_t uxLength;
+	const IPPacket_t * pxIPPacket;
+	UBaseType_t uxIPHeaderLength;
+	ProtocolPacket_t *pxProtPack;
+	uint8_t ucProtocol;
+	uint16_t usLength;
+	uint16_t ucVersionHeaderLength;
+	BaseType_t xLocation = 0;
+	size_t uxMinimumLength;
+	BaseType_t xResult = pdPASS;
+
+		do
+		{
+			/* Check for minimum packet size: Ethernet header and an IP-header, 34 bytes */
+			if( uxBufferLength < sizeof( IPPacket_t ) )
+			{
+				xResult = pdFAIL;
+				xLocation = 1;
+				break;
+			}
+
+			/* Parse the packet length. */
+			pxIPPacket = ipPOINTER_CAST( const IPPacket_t *, pucEthernetBuffer );
+
+			/* Per https://tools.ietf.org/html/rfc791, the four-bit Internet Header
+			Length field contains the length of the internet header in 32-bit words. */
+			ucVersionHeaderLength = pxIPPacket->xIPHeader.ucVersionHeaderLength;
+			if( ( ucVersionHeaderLength < 0x45U ) || ( ucVersionHeaderLength > 0x4FU ) )
```

Comment:
Can we make 0x45U and 0x4FU something more descriptive?

Reply:
+1, the `0x40` upper nibble indicates IPv4 and the [`0x05-0x0f` lower nibble are the valid values for the IHL](https://en.wikipedia.org/wiki/IPv4#Header).  There should be constants or `#defines` for these somewhere.  Also, the comment on lines 1939-1940 only mention length but we are also validating version here as well.

Reply:
I propose these two macro's for this purpose:
~~~c
/* The first byte in the IPv4 header combines the IP version (4) with
with the length of the IP header. */
#define	ipIPV4_VERSION_HEADER_LENGTH_MIN	0x45U
#define	ipIPV4_VERSION_HEADER_LENGTH_MAX	0x4FU
~~~

> only mention length but we are also validating version here as well.

What about the following?
~~~C
    /* Test if the length of the IP-header is between 20 and 60 bytes,
    and if the IP-version is 4. */
    else if( ( pxIPHeader->ucVersionHeaderLength < ipIPV4_VERSION_HEADER_LENGTH_MIN ) ||
             ( pxIPHeader->ucVersionHeaderLength > ipIPV4_VERSION_HEADER_LENGTH_MAX ) )
~~~